lib/ecto/repo.ex

```diff
@@ -657,6 +666,64 @@ defmodule Ecto.Repo do
               opts :: Keyword.t()
             ) :: Ecto.Schema.t()
 
+  @doc """
+  Reloads a given schema or schema list from the database.
+
+  When using with lists, ordering is guaranteed to be kept. Results not found in
+  the database will be returned as `nil`.
+  
+  ## Options
+
+    * `:prefix` - The prefix to run the query on (such as the schema path
+      in Postgres or the database in MySQL). This will be applied to all `from`
+      and `join`s in the query that did not have a prefix previously given
+      either via the `:prefix` option on `join`/`from` or via `@schema_prefix`
+      in the schema. For more information see the "Query Prefix" section of the
+      `Ecto.Query` documentation.
```

Comment:
I don't think we should accept this option, rather, we should use the prefix from the structs themselves. You can find it under the `__meta__` field.

Reply:
You can test this by using `Ecto.put_meta(struct, :prefix, "another")` in test/ecto/repo_test.exs and then using assert_receive to check to which prefix the query went to. :)

If that doesn't work, have an integration test that sets `Ecto.put_meta(struct, :prefix, "invalid")` and try to reload it and see that it will fail.